### PR TITLE
Fix `GROUP BY` tests grouping over `MISSING`

### PR DESCRIPTION
**Pull request by alancai98**
Addresses #73.

`partiql-lang-kotlin` and tests ported from `partiql-lang-kotlin` to `partiql-tests` had the incorrect behavior when a group key value was `MISSING`. `MISSING` should be coerced to `NULL` as per [section 11.1.1](https://partiql.org/assets/PartiQL-Specification.pdf#subsubsection.11.1.1) of the PartiQL spec. Previously, the group key would be `MISSING` in the edge case of when no other group keys values were `NULL`.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: partiql-tests-data/eval/query/group-by/group-by.ion [hunk 1]
```diff
@@ -1621,68 +1622,7 @@
...
-    name:"SELECT VALUE { 'groupExp': groupExp } FROM simple_1_col_1_group GROUP BY MISSING AS groupExp",
-    statement:"SELECT VALUE { 'groupExp': groupExp } FROM simple_1_col_1_group GROUP BY MISSING AS groupExp",
-    assert:{
-      result:EvaluationSuccess,
-      evalMode:[
-        EvalModeCoerce,
-        EvalModeError
-      ],
-      output:$bag::[
-        {
```
Comment (alancai98) [inline partiql-tests-data/eval/query/group-by/group-by.ion:1685]: These ported tests were nearly identical to the previous tests other than the group key alias. Removed in this PR due to this redundancy.

File: partiql-tests-data/eval/query/group-by/group-by.ion [hunk 2]
```diff
@@ -1832,6 +1776,7 @@
           supplierId_mixed:11
         },
         {
+          supplierId_mixed:null  // missing coerced to null as per 11.1.1
```
Comment (alancai98) [inline partiql-tests-data/eval/query/group-by/group-by.ion:1779]: This was actually a bug in the Kotlin implementation (tracking issue: https://github.com/partiql/partiql-lang-kotlin/issues/1014). Basically depending on the order of the `FROM` source elements, the group by key name was a different value (missing or null).

**Review by vgapeyev — APPROVED** [review]
Looks all right.
